### PR TITLE
BNB Smart Chain

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

**- BNB hidden under debug mode**
- `coins-eth` is regenered automatically from `submodules`
- added `T3T1` support for EVM coins
- soon I will update fw repo to have correct EVM defs

**Bugs**
- Native currency should be `BNB`, not `BSC`. Networks config has to be extended
- Blockbook should also report network, not just native currency symbol
- old internal txs are missing, we have to hide chart or sync the rest
  - number of all txs and incoming amount as it can be incorrect
- Balance will be always correct. However, we should introduce a banner which informs users that old internal txs before xxxx block are missing
Screenshot 2024-06-13 at 13 19 59
Screenshot 2024-06-29 at 22 07 03

## Screenshots:
Screenshot 2024-06-29 at 21 59 44
Screenshot 2024-06-29 at 21 59 35
Screenshot 2024-06-29 at 21 59 18

image

### CODE REVIEW

**Review by jvaclavik — APPROVED** [review]

File: suite-common/wallet-config/src/networksConfig.ts [hunk 1]
```diff
@@ -351,6 +351,27 @@ export const networks = {
...
+        chainId: 56,
+        bip43Path: "m/44'/60'/0'/0/i",
+        decimals: 18,
+        testnet: false,
+        explorer: {
+            tx: 'https://staging-bsc.trezor.io/tx/',
+            account: 'https://staging-bsc.trezor.io/address/',
+            nft: 'https://staging-bsc.trezor.io/nft/',
+            address: 'https://staging-bsc.trezor.io/address/',
+            queryString: '',
```
Comment (tomasklim) [inline suite-common/wallet-config/src/networksConfig.ts]: TODO

File: packages/connect-common/files/coins-eth.json [hunk 2]
```diff
@@ -1,98 +1,149 @@
...
-        "url": "https://ethereum.org"
-    },
-    {
-        "blockchain_link": {
-            "type": "blockbook",
-            "url": ["https://sepolia1.trezor.io"]
+        {
+            "blockchain_link": {
+                "type": "blockbook",
+                "url": ["https://blockbook-dev2.corp.sldev.cz:9165"]
```
Comment (tomasklim) [inline packages/connect-common/files/coins-eth.json]: TODO

File: packages/blockchain-link/src/ui/config.ts [hunk 3]
```diff
@@ -87,6 +87,28 @@ export default [
             subscribe: '0x2a0c0dbecc7e4d658f48e01e3fa353f44050c208', // Poloniex: https://www.walletexplorer.com/wallet/Poloniex.com
         },
     },
+    {
+        blockchain: {
+            name: 'BNB Smart Chain',
+            worker: 'js/blockbook-worker.js',
+            server: ['https://blockbook-dev2.corp.sldev.cz:9165'],
```
Comment (tomasklim) [inline packages/blockchain-link/src/ui/config.ts]: TODO

**Review by tomasklim — COMMENTED** [review]
Change before merge

**Review by jvaclavik — APPROVED** [review]

File: packages/connect/src/backend/Blockchain.ts [hunk 4]
```diff
@@ -120,7 +120,9 @@ export class Blockchain {
 
         // There is no `rippled` setting that defines which network it uses neither mainnet or testnet
         // see: https://xrpl.org/parallel-networks.html
-        const shortcut = this.coinInfo.shortcut === 'tXRP' ? 'XRP' : this.coinInfo.shortcut;
+        let shortcut = this.coinInfo.shortcut === 'tXRP' ? 'XRP' : this.coinInfo.shortcut;
+
+        shortcut = shortcut.toLowerCase() === 'bnb' ? 'bsc' : shortcut;
```
Comment (mroz22) [inline packages/connect/src/backend/Blockchain.ts]: don't we control the shortcut? why is it?

Comment (tomasklim) [reply packages/connect/src/backend/Blockchain.ts]: Blockbook uses shortcut `BNB` because `bnb` is native currency, but we try to solve it

**Comment by bosomt** [comment]
**QA OK**

known bugs:
- device says BNB bud app says BSC
- missing internal transactions so graph is already bellow 0 values

**tx sent** 
https://bsc1.trezor.io/tx/0xc6844bc87343a78363c353be27a7c7a9a3cdb633468f68a24c76242db340704b
**token sent**
https://bsc1.trezor.io/tx/0x31ef5a8cff84f291c2086ee7d3e7852d7df4b4efef0d9267f691664b5af873bd

**Info:**
 - Suite version: desktop 24.7.1 (3856b5e5f845e68f754487dee863c6ea0c347070)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.7.1 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.0 regular (revision dd4671a5104952ef505d28d1f9e94d1484b4607a)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**